src/components/progress-indicators/Shimmer.tsx

```diff
@@ -0,0 +1,77 @@
+import * as React from 'react'
+import { View } from 'react-native'
+import LinearGradient from 'react-native-linear-gradient'
+import Animated, { useAnimatedStyle, useSharedValue, withRepeat, withSequence, withTiming } from 'react-native-reanimated'
+
+import { useHandler } from '../../hooks/useHandler'
+import { cacheStyles, Theme, useTheme } from '../services/ThemeContext'
+
+type Props = {
+  children?: React.ReactNode
+  visible?: boolean
+}
+
+export const Shimmer = (props: Props) => {
+  const { children = null, visible = true } = props
+
+  const theme = useTheme()
+  const styles = getStyles(theme)
+
+  const offset = useSharedValue(-1000)
+
+  const startAnimation = useHandler(() => {
+    const duration = 1000
+    offset.value = withRepeat(withSequence(withTiming(-1000, { duration }), withTiming(0, { duration })), -1, false)
+  })
+
+  React.useEffect(() => {
+    startAnimation()
+  }, [startAnimation])
+
+  const animStyle = useAnimatedStyle(() => {
+    return {
+      transform: [{ translateX: offset.value }]
+    }
+  })
+
+  return (
+    <View>
+      <View style={{ opacity: visible ? 0 : 1 }}>{children}</View>
+      {visible ? (
```

Comment:
I'd prefer the Shimmer component act like a Spinner so it doesn't take any children but just renders as normal and gets its size from the parent like a regular View. This make it an easy replacement for regular spinners.